### ISSUE TITLE
[Menu] Accessibility audit

### ISSUE THREAD

**Issue by DanielleRameau** [sl-design-system/components#2574]
### Provide a general summary of the issue here

Test menu component on accessibility.

### 👤 Your name

DR

### 🧢 Your Product/Team

SLDS

**Comment by KatarzynaWM** [comment]
**testing environment and tools:**

Windows 11
Chrome version 140.0.7339.81
screen reader: NVDA
Color Contrast Analyser from TPGi

--------
For variant with Keyboard shortcuts: currently screen reader user is not informed, that part of the read text refers to a keyboard shortcut. Some kind of visually hidden label is needed to pass this information.

<img width="1132" height="640" alt="Image" src="https://github.com/user-attachments/assets/51b00c7f-e1a8-4afe-900f-c00a1778b45c" />

**Comment by KatarzynaWM** [comment]
Submenu variant:
Before the first opening of the submenu, the menu item lacks aria-expanded attribute (so the screen reader user is not informed, that there is a submenu there):

<img width="1206" height="594" alt="Image" src="https://github.com/user-attachments/assets/b139bcba-86e7-4828-a78c-87ae13a2ac94" />

The attribute appears only after first opening of the submenu:

<img width="1366" height="701" alt="Image" src="https://github.com/user-attachments/assets/649df1da-a84e-488c-a2a9-45a432dd1ec0" />

**Comment by KatarzynaWM** [comment]
I have doubts on how disabled elements are handled. Right now, there are not announced by screen reader (which is ok), but they are counted within the total number of existing elements. So, the user can guess that there is something there, but can't get the information, what is it (it can be reached by changing reading mode though). Maybe disabled elements shouldn't be "counted"?

<img width="1283" height="624" alt="Image" src="https://github.com/user-attachments/assets/84f9fa6a-d188-48cd-a873-c3b4793ed8bd" />
